src/components/MainMenu/NavDropdown.tsx

```diff
@@ -0,0 +1,111 @@
+import classNames from "classnames";
+import * as React from "react";
+import { Link } from "react-router-dom";
+
+import {
+  generateCategoryUrl,
+  generateCollectionUrl,
+  generatePageUrl
+} from "../../core/utils";
+import {
+  SecondaryMenu_shop_navigation_secondary_items,
+  SecondaryMenu_shop_navigation_secondary_items_children
+} from "../Footer/types/SecondaryMenu";
+import { OverlayContext, OverlayTheme, OverlayType } from "../Overlay/context";
+import NavItem from "./NavItem";
+import { MainMenu_shop_navigation_main_items } from "./types/MainMenu";
+import { MainMenuSubItem } from "./types/MainMenuSubItem";
+
+import "./scss/index.scss";
+
+export const generateNavLink = (
```

Comment:
Can we make this a component and change its signature, so we can use it as `<NavLink item={navItem} {...linkProps} />`?